### ISSUE TITLE
loading sparse matrix with only null values in v4 format

### ISSUE THREAD

**Issue by SciCed** [tbeu/matio#250]
It is no longer possible to load a sparse matrix that contains only null values in v4 format.

This zip file contains the matfile.
[spds.zip](https://github.com/user-attachments/files/15821660/spds.zip)

**Comment by tbeu** [comment]
Thanks for reporting? What exatly do you meean by "no longer"? Do you have a reference version where it worked last? Did you run git bisect to locate the bad commit? Could you please also provide the code snippet how to generate the file such that I can add it to the acceptance tests? Thansk again.

**Comment by SciCed** [comment]
We are upgrading from Matio 1.5.9 to 1.5.27 which is a big step :)
I did not use git bisect, but I investigated the code, and if I've well understood the v4 format, sparse data are stored in a "table" with one line by non-zero values and an extra line that contains the sparse dimension.

The condition on dimensions is now checking that there are at least two rows, that is not true in case of sparse that contains only null values.
see mat4.c:368 in 1.5.27
```
if ( matvar->dims[0] < 2 ) {
    return MATIO_E_FILE_FORMAT_VIOLATION;
}
```

I think that commit introduced the bug https://github.com/tbeu/matio/commit/aefd09e9d20aebb4a0b0a5e5c619f46038dba3bc

The file has been generated by Scilab, which does not use matio to generate mat files in v4 format.
Here is the Scilab code:
```
spbs = sparse(0);
file = TMPDIR + "/spbs.mat";
savematfile(file, "-v4", "spbs");
```
